subsys/net/lib/shell/iface.c

```diff
@@ -817,6 +818,47 @@ static int cmd_net_default_iface(const struct shell *sh, size_t argc, char *argv
 	return 0;
 }
 
+static int cmd_net_link_speed(const struct shell *sh, size_t argc, char *argv[])
+{
+#if !defined(CONFIG_NET_L2_ETHERNET_MGMT)
+       PR_WARNING("Unsupported command, please enable CONFIG_NET_L2_ETHERNET_MGMT\n");
+       return -ENOEXEC;
+#else
+       struct ethernet_req_params params = { 0 };
+       int idx = get_iface_idx(sh, argv[1]);
+       uint16_t speed = atoi(argv[2]);
+       struct net_if *iface;
+
+       if (idx < 0) {
+               return -ENOEXEC;
+       }
+
+       iface = net_if_get_by_index(idx);
+       if (!iface) {
+               PR_WARNING("No such interface in index %d\n", idx);
+               return -ENODEV;
+       }
+
+       switch (speed) {
+       case 10:
+               params.l.link_10bt = true;
+               break;
+       case 100:
+               params.l.link_100bt = true;
+               break;
+       case 1000:
```

Comment:
We support also 2.5G and 5G link speeds atm so those should be checked too here.
The `struct ethernet_config` in `include/zephyr/net/ethernet.h` does not contain link speed setting for those, so please add a commit in this PR that sets 2.5 and 5 G speeds (I am referring to anonymous struct named `l` here).